interactions/client/get.py

```diff
@@ -0,0 +1,394 @@
+        return _search_iterable(*args, **kwargs)
+
+
+async def _http_request(
+    obj: Type[_A],
+    http: HTTPClient,
+    request: Union[Coroutine, List[Union[_A, Coroutine]], List[Coroutine]] = None,
+    _name: str = None,
+    **kwargs,
+) -> Union[_A, List[_A]]:
+
+    if not request:
+        if obj in (Role, Emoji):
+            _guild = Guild(**await http.get_guild(kwargs.pop("guild_id")), _client=http)
+            _func = getattr(_guild, _name)
+            return await _func(**kwargs)
+
+        _func = getattr(http, _name)
+        _obj = await _func
+        return obj(**_obj, _client=http)
+
+    if not isinstance(request, list):
+        return obj(**await request, _client=http)
+
+    return [obj(**await req, _client=http) if isawaitable(req) else req for req in request]
+
+
+async def _return_cache(
+    obj: Union[Optional[_A], List[Optional[_A]]]
+) -> Union[Optional[_A], List[Optional[_A]]]:
+    await sleep(0)  # iirc Bluenix meant that any coroutine should await at least once
+    return obj
+
+
+def _get_cache(
+    _object: Type[_A], client: Client, kwarg_name: str, _list: bool = False, **kwargs
+) -> Union[Optional[_A], List[Optional[_A]]]:
+
+    if _list:
+        _obj = []
+        if isinstance(_object, Member):  # Can't be more dynamic on this
+            _values = ()
+            _guild_id = Snowflake(kwargs.get("guild_id"))
+            for _id in kwargs.get("member_ids"):
+                _values += (
+                    (
+                        _guild_id,
+                        Snowflake(_id),
+                    ),
+                )
+            _obj.extend(client.cache[_object].get(item) for item in _values)
+
+        else:
+            _obj.extend(
+                client.cache[_object].get(Snowflake(_id), None) for _id in kwargs.get(kwarg_name)
+            )
+    else:
+        if isinstance(_object, Member):
+            _values = (
+                Snowflake(kwargs.get("guild_id")),
+                Snowflake(kwargs.get("member_id")),
+            )  # Fuck it, I can't be dynamic on this
+        else:
+            _values = Snowflake(kwargs.get(kwarg_name))
+
+        _obj = client.cache[_object].get(_values)
+    return _obj
+
+
+def _search_iterable(items: Iterable[_A], **kwargs) -> Optional[_A]:
+
+    if not isinstance(items, Iterable):
+        raise LibraryException(message="The specified items must be an iterable!", code=12)
+
+    if not kwargs:
+        raise LibraryException(
+            message="You have to specify either a custom check or a keyword argument to check against!",
+            code=12,
+        )
+
+    if len(list(kwargs)) > 1:
+        raise LibraryException(
+            message="Only one keyword argument to check against is allowed!", code=12
+        )
+
+    _arg = str(list(kwargs)[0])
+
+    __obj = next(
+        (
+            item
+            for item in items
+            if (
+                str(getattr(item, _arg, None)) == str(kwargs.get(_arg))
```

Comment:
This logically reads out to: `True if not True else foo`. What's the goal from the object iteration if the iterable can be any expression?

Reply:
I don't get what you mean, sorry